server/src/main/java/org/elasticsearch/index/mapper/TermBasedFieldType.java

```diff
@@ -55,15 +60,33 @@ public boolean mayExistInIndex(SearchExecutionContext context) {
 
     @Override
     public Query termQuery(Object value, SearchExecutionContext context) {
-        failIfNotIndexed();
-        return new TermQuery(new Term(name(), indexedValueForSearch(value)));
+        if (allowDocValueBasedQueries()) {
+            failIfNotIndexedNorDocValuesFallback(context);
+        } else {
+            failIfNotIndexed();
+        }
+        if (isIndexed()) {
+            return new TermQuery(new Term(name(), indexedValueForSearch(value)));
+        } else {
+            // Should we use SortedSetDocValuesField.newSlowExactQuery instead?
+            return new DocValuesTermsQuery(name(), indexedValueForSearch(value));
```

Comment:
@jpountz thoughts on which query to use?

Reply:
Use `SortedSetDocValuesField.newSlowExactQuery`, `DocValuesTermsQuery` is optimized for querying multiple terms and would have a worse runtime for a single term.